tests/api/api.spec.ts

```diff
@@ -491,6 +496,163 @@ describe("@dataform/api", () => {
         "\nselect * from (\nselect * from `tada-analytics.df_integration_test.sample_data`)\nwhere true"
       );
       expect(exampleUsingInline.dependencies).includes("sample_data");
+
+      // Check view
+      expect(tableNames).includes("example_view");
+      const exampleView = graph.tables.filter(t => t.name == "example_view")[0];
+      expect(exampleView.type).equals("view");
+      expect(exampleView.query).equals(
+        "\nselect * from `tada-analytics.df_integration_test.sample_data`"
+      );
+      expect(exampleView.dependencies).deep.equals(["sample_data"]);
+
+      // Check table
+      expect(tableNames).includes("example_table");
+      const exampleTable = graph.tables.filter(t => t.name == "example_table")[0];
+      expect(exampleTable.type).equals("table");
+      expect(exampleTable.query).equals(
+        "\nselect * from `tada-analytics.df_integration_test.sample_data`"
+      );
+      expect(exampleTable.dependencies).deep.equals(["sample_data"]);
+
+      // Check sample data
+      expect(tableNames).includes("sample_data");
+      const exampleSampleData = graph.tables.filter(t => t.name == "sample_data")[0];
+      expect(exampleSampleData.type).equals("view");
+      expect(exampleSampleData.query).equals(
+        "select 1 as sample union all\nselect 2 as sample union all\nselect 3 as sample"
+      );
+      expect(exampleSampleData.dependencies).to.eql([]);
+    });
+
+    it("bigquery using v2 language compiles", async () => {
+      const graph = await compile({ projectDir: path.resolve("df/examples/bigquery_language_v2") });
+
+      // throw new Error(graph.graphErrors.compilationErrors[0].message);
```

Comment:
Doesn't look quite right

Reply:
On this note, I think we probably need a few test cases for failures as we don't currently have any and added a lot of new checks!
Let's discuss validation/error handling in general though

Reply:
removed the commented out code and added a bunch of error test cases!